src/main/java/shef/data/Trending.java

```diff
@@ -14,6 +14,35 @@
 
 package shef.data;
 
+import javax.servlet.http.HttpServletRequest;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.CompositeFilter;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.PreparedQuery;
+
 public class Trending implements RecipeFilter {
-  
-}
\ No newline at end of file
+
+  private DatastoreService datastore;
+  private static final int MIN_LIKES = 10;
+
+  public Trending(HttpServletRequest request) {
```

Comment:
Why does the request get passed in here?

Reply:
Eventually the trending class will need to look at parameters from the request (like the minimum number of likes or something), but this functionality hasn't been implemented yet.

Reply:
IMO, it would be better to pass in only those things required from the request, rather than the request itself. If need be, you can have another data structure to extract and store the required fields from request.

Reply:
Since I don't know what exact parameters it'll need yet, I'll just remove request from the constructor.